src/lucky_record/associations.cr

```diff
@@ -9,9 +9,20 @@ module LuckyRecord::Associations
 
   macro belongs_to(type_declaration)
     {% assoc_name = type_declaration.var }
-    {% model = type_declaration.type %}
 
-    field {{ assoc_name.id }}_id : Int32
+    {% if type_declaration.type.is_a?(Union) %}
+      {% model = type_declaration.type.types.first %}
+      {% nilable = true %}
+    {% else %}
+      {% model = type_declaration.type %}
+      {% nilable = false %}
+    {% end %}
+
+    {% if nilable %}
+      field {{ assoc_name.id }}_id : Int32?
+    {% else %}
+      field {{ assoc_name.id }}_id : Int32
+    {% end %}
 
     def {{ assoc_name.id }}
       {{ model }}::BaseQuery.new.find({{ assoc_name.id }}_id)
```

Comment:
Instead of adding `find(id: Nil)` and returning `nil`, we should check if the id is present and then make the query:

```crystal
def {{ assoc_name.id }}
  {{ assoc_name.id }}_id.try do |value|
    {{ model }}::BaseQuery.new.find({{ assoc_name.id }}_id)
  end
```

The reason is that `find` should not accept `nil` at all. I'll need to change that later on. I'm planning to rework how types are handled in queries so it might be a bit until I can tackle that